Source/RapyutaSimulationPlugins/Private/Core/RRUObjectUtils.cpp

```diff
@@ -0,0 +1,144 @@
+#include "Core/RRUObjectUtils.h"
+
+// UE
+#include "Components/ActorComponent.h"
+#include "GameFramework/Actor.h"
+
+// RapyutaSimulationPlugins
+#include "Core/RRBaseActor.h"
+#include "Core/RRMathUtils.h"
+
+void URRUObjectUtils::SetupActorTick(AActor* InActor, bool bIsTickEnabled, float InTickInterval)
+{
+    // Tick if not required could be disabled to improve performance
+    InActor->PrimaryActorTick.bCanEverTick = bIsTickEnabled;
+    InActor->PrimaryActorTick.bStartWithTickEnabled = bIsTickEnabled;
+    InActor->SetActorTickEnabled(bIsTickEnabled);
+    if (bIsTickEnabled)
+    {
+        InActor->PrimaryActorTick.Target = InActor;
+    }
+    if (InTickInterval > 0.f)
+    {
+        InActor->PrimaryActorTick.TickInterval = InTickInterval;
+    }
+
+#if RAPYUTA_SIM_DEBUG
+    // These are for troubleshooting only, in case Tick does not work even having been enabled.
+    // These also don't apply the same for all actors
+    InActor->PrimaryActorTick.bAllowTickOnDedicatedServer = bIsTickEnabled;
+    InActor->PrimaryActorTick.bHighPriority = bIsTickEnabled;
+    InActor->PrimaryActorTick.bRunOnAnyThread = bIsTickEnabled;
+    InActor->PrimaryActorTick.RegisterTickFunction(InActor->GetLevel());
+    // --> Call Child component's RegisterAllComponentTickFunctions(bRegister);
+    InActor->RegisterAllActorTickFunctions(bIsTickEnabled, bIsTickEnabled);
+#endif
+}
+
+void URRUObjectUtils::SetupComponentTick(UActorComponent* InComponent, bool bIsTickEnabled)
+{
+    InComponent->PrimaryComponentTick.bCanEverTick = bIsTickEnabled;
+    InComponent->PrimaryComponentTick.bStartWithTickEnabled = bIsTickEnabled;
+    InComponent->bTickInEditor = bIsTickEnabled;
+}
+
+void URRUObjectUtils::SetupDefaultRootComponent(AActor* InActor)
```

Comment:
Just from curiosity, do you know why UE has two diifferent methods for functions inside/outside of constructors?
cc @chalice-graeme

Reply:
One of the most peculiar traits of UE that I am not a fan of (and thus do not favor BP dev at all), but I think it might be to support child BP class instantiation, which requires member objects to be created as default objects as inside Ctor (so they could be configured/modified in the BP Editor) vs BeginPlay, where only NewObject is used.